src/elements/EditableTable.tsx

```diff
@@ -0,0 +1,303 @@
+        const sum =
+            numericVals.length > 0 ? numericVals.reduce((s: number, a: number) => s + a, 0) : 0;
+        total[col.name] = sum;
+    }
+    return total;
+};
+
+/**
+ * A table form element.
+ * @displayName Editable Table
+ * @description A table form element that allows editing.
+ * @param props The props that will be provided by the Workflow runtime.
+ */
+function EditableTableElement(props: EditableTableElementProps): React.ReactElement | null {
+    const {
+        rows,
+        cols,
+        maxHeight,
+        maxWidth,
+        onMouseEnter,
+        onMouseLeave,
+        onChange,
+        raiseEvent,
+        setValue,
+        size,
+        editTooltip,
+        saveTooltip,
+        stickyHeader,
+        footerLabel,
+        footerStyle,
+        saveIcon,
+        editIcon,
+    } = props;
+    const [data, setData] = useState(rows);
+    setValue(data);
+    useEffect(() => {
+        setData(rows);
+    }, [rows]);
+
+    const includeTotals = cols.some((x) => x.includeTotal);
+    const totalData: Total = calculateTotals(cols, rows);
+    const footerLbl = footerLabel ?? "Total";
+    const editIconSrc: string = editIcon ?? "edit";
+    const saveIconSrc: string = saveIcon ?? "save";
+    const editTitle = editTooltip ?? "Edit";
+    const saveTitle = saveTooltip ?? "Save";
+
+    const handleEditToggleClick = (event: React.MouseEvent, row: Row) => {
+        if (row.isEditing) {
+            raiseEvent("custom", {
+                eventType: "rowChange",
+                rowData: { rows: data, row: row },
+            });
+        }
+        row.isEditing = !row.isEditing;
+        setData([...data]);
+        setValue(data);
+    };
+
+    const handleMouseEnter = (event: React.MouseEvent, row: Row, column: string) => {
+        raiseEvent("custom", {
+            eventType: "mouseEnter",
+            rowData: { rows: data, row: row, column: column },
+        });
+        onMouseEnter?.(row);
+    };
+
+    const handleMouseLeave = (event: React.MouseEvent, row: Row, column: string) => {
+        raiseEvent("custom", {
+            eventType: "mouseLeave",
+            rowData: { rows: data, row: row, column: column },
+        });
+        onMouseLeave?.(row);
+    };
+
+    const handleCellChange = (
+        event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+        row: Row,
+        column: Column
+    ) => {
+        if ((event.target as any).type === "number") {
+            const regex = /^[0-9\b]+$/;
+            if (event.target.value === "" || regex.test(event.target.value)) {
+                row[column.name] = Number(event.target.value);
+            }
+            row[column.name] = Number(event.target.value);
+        } else if (column?.parse) {
+            row[column.name] = column.parse(event, row, column);
+        } else {
+            row[column.name] = event.target.value;
+        }
+        if (onChange) {
```

Comment:
If the change is effectively done when the cell changes, then what do we need the Save button for? It just raises the same event again and toggles the button state.

Reply:
Maybe I should do away with the edit toggle and just have rows that are editable or not.  It does mean fewer clicks for the user that way and simplifies the form element.